dotcom-rendering/src/web/server/articleToHtml.tsx

```diff
@@ -207,16 +207,23 @@ export const articleToHtml = ({ article: CAPIArticle }: Props): string => {
 		),
 	);
 
-	const hasAmpInteractiveTag = CAPIArticle.tags.some(
-		(tag) => tag.id === 'tracking/platformfunctional/ampinteractive',
-	);
+	const getAmpLink = (tags: TagType[]) => {
+		// We don’t support AMP for Immersive articles in frontend
+		// https://github.com/guardian/frontend/blob/149c8d3be273edf784465a780ee332bd58e2a9b3/common/app/model/content.scala#L95
+		if (CAPIArticle.format.display === 'ImmersiveDisplay') return undefined;
```

Comment:
I feel as though ideally if the decision is being made in Frontend then we'd probably want to add a property before it gets to DCR saying `doNotRenderAmpLink` or similar.

But given that there was already a check for `InteractiveDesign` here, it would be a bigger refactor to move the logic to Frontend, so this looks good to me.

And I think moving the logic into the `getAmpLink` function really helps readability here 👍